cache/lru/lru.go

```diff
@@ -0,0 +1,44 @@
+package lru
+
+import (
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// Cache encapsulates a thread-safe fixed size LRU cache
+// as defined in hashicorp/golang-lru.
+type Cache struct {
+	lru *lru.Cache
+}
+
+// New returns a new LRU cache.
+func New(size int) (*Cache, error) {
+	lruCache, err := lru.New(size)
+	if err != nil {
+		return nil, err
+	}
+	return &Cache{lru: lruCache}, nil
+}
+
+// Get executes a lookup and returns whether a key exists in the cache along with and its value.
+func (c *Cache) Get(key string) (interface{}, bool, error) {
+	value, ok := c.lru.Get(key)
+	return value, ok, nil
+}
+
+// Purge evicts all keys present in the cache.
+func (c *Cache) Purge() error {
+	c.lru.Purge()
+	return nil
+}
+
+// Remove evicts a specific key from the cache.
+func (c *Cache) Remove(key string) error {
+	c.lru.Remove(key)
+	return nil
+}
+
+// Set registers a key-value pair to the cache.
+func (c *Cache) Set(key string, value interface{}) error {
+	c.lru.Add(key, value)
+	return nil
+}
```

Comment:
what about SetTTL ?
This means struct will not be able to implement the interface ...

Reply:
The golang-lru package does not support TTL expiry by default. I tried to circumvent it by [using `time.AfterFunc`](https://github.com/beatlabs/patron/pull/138#discussion_r368234581), but it was not deemed a good option.

There's an idea of [breaking the interface up](https://github.com/beatlabs/patron/pull/138#discussion_r368234670), to support TTL or not, but I haven't decided on what's best. 

Any recommendations are welcome!